### PR TITLE
chore(docs): add environment related constants

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
Add environment related constants to docs (related to [this comment](https://github.com/aws-amplify/amplify-ui/pull/2354#discussion_r931666984))

#### Issue #, if available
NA

#### Description of how you validated changes
Manual testing
#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/utils/environment.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export const isDev = process.env.NODE_ENV === 'development';
+export const isProd = process.env.NODE_ENV === 'production';
+export const isTest = process.env.NODE_ENV === 'test';
```
Comment (0618) [inline docs/src/utils/environment.ts:3]: Looks like you added `FF_REACT_NATIVE` in https://github.com/aws-amplify/amplify-ui/pull/2354/files, where did we get `NODE_ENV` or it's some sort of default variable?

**Comment by 0618** [comment]
Why do we open this PR instead of updating https://github.com/aws-amplify/amplify-ui/pull/2354?

File: docs/src/utils/environment.ts (hunk 1 above)
Comment (calebpollman) [reply docs/src/utils/environment.ts:3]: The `isDev` flag no longer has relevancy to https://github.com/aws-amplify/amplify-ui/pull/2354 so i moved it in to its own PR. Added `isProd` and `isTest` so they would already be there when needed in the future

Comment (calebpollman) [reply docs/src/utils/environment.ts:3]: For further context you can read [this comment](https://github.com/aws-amplify/amplify-ui/pull/2354#discussion_r931666984).

**Review by ioanabrooks — APPROVED** [review]

**Review by reesscot — APPROVED** [review]
LGTM

**Review by zchenwei — APPROVED** [review]

**Pull request merged**